src/main.py

```diff
@@ -1,163 +1,31 @@
-        # fraction_of_articles_to_use=1/150,
+def update_sql_and_pinecone_dbs():
+    updater = ARDUpdater(
+        min_tokens_per_block=200,
+        max_tokens_per_block=300,
     )
-    dataset.get_alignment_texts()
-    
-    print(len(dataset.embedding_strings))
-    print(dataset.total_word_count)
-    print(dataset.total_block_count)
-    print(dataset.articles_count)
-    
-    dataset.get_embeddings()
-    dataset.save_data()
-    
-@retry(wait=wait_random_exponential(min=1, max=20), stop=stop_after_attempt(4))
-def get_embedding(text: str) -> np.ndarray:
-    result = openai.Embedding.create(model=EMBEDDING_MODEL, input=text)
-    return np.array(result["data"][0]["embedding"])
-
-def print_out_dataset_stuff():
-    with open(PATH_TO_DATASET_PKL, 'rb') as f:
-        dataset = pickle.load(f)
-
-    embeddings_len = len(dataset.embedding_strings)
-    i1 = random.randint(0,embeddings_len-1)
-    #i2 = random.randint(0,embeddings_len-1)
-    #print(len(dataset.embeddings))
-    #print(len(dataset.embedding_strings))
-    #embedding_test = get_embedding(dataset.embedding_strings[i])
-    #print(np.dot(embedding_test,dataset.embeddings[i]))
-
-    metadata_i1 = dataset.embeddings_metadata_index[i1]
-    print("metadata:",dataset.metadata[metadata_i1])
-    print("embedding_string:",dataset.embedding_strings[i1])
-    #print("embedding_vector:",dataset.embeddings[i1])
-    embedding_of_string1 = get_embedding(dataset.embedding_strings[i1])
-
-    #metadata_i2 = dataset.embeddings_metadata_index[i2]
-    #print("metadata:",dataset.metadata[metadata_i2])
-    #print("embedding_string:",dataset.embedding_strings[i2])
-    #print("embedding_vector:",dataset.embeddings[i1])
-    #embedding_of_string2 = get_embedding(dataset.embedding_strings[i2]) 
-    #embedding_of_string2 = get_embedding("000000000000000000000000000000000000000000000000000000000000000000000000000000000")
-
-    #print(len(embedding_of_string1))
-    vector = dataset.embeddings[i1]
-    plot_likelihood(vector)
-    #plot_likelihood(get_embedding("tst"))
-    print(max(vector), min(vector))
-    print(sum([x**2 for x in vector]))
-
-
-    
-
-    #print(np.dot(embedding_of_string1, embedding_of_string2))
-
-def plot_likelihood(embeddings, num_buckets=200):
-    # Calculate the histogram
-    histogram, bin_edges = np.histogram(embeddings.flatten(), bins=num_buckets, range=(embeddings.min(), embeddings.max()))
-
-    # Normalize the histogram to get likelihoods
-    likelihoods = histogram / embeddings.flatten().size
-
-    # Plot the likelihoods
-    plt.bar(bin_edges[:-1], likelihoods, width=(bin_edges[1] - bin_edges[0]), edgecolor="k", alpha=0.7)
-    plt.xlabel("Value")
-    plt.ylabel("Likelihood")
-    plt.title("Likelihood of Floats in the Vector Embedding")
-    plt.savefig("bla.png")
+    if RESET_DBS:
+        updater.reset_dbs()
+    updater.update(['gwern_blog'])
 
-    
-    
 
+def main():
+    update_sql_and_pinecone_dbs()
```

Comment:
why not just directly call this function, rather than going through this `main()` call?

Reply:
Because I was told that it's more modular and respects norms/conventions, but I realize it doesn't matter in this case. I don't mind either way, and can change it back